src/librustc/ty/mod.rs

```diff
@@ -1578,12 +1579,36 @@ impl<'a, 'gcx, 'tcx> AdtDef {
         }
     }
 
-    pub fn destructor(&self) -> Option<DefId> {
-        self.destructor.get()
+    pub fn destructor(&self, tcx: TyCtxt<'a, 'gcx, 'tcx>) -> Option<DefId> {
+        if self.flags.get().intersects(AdtFlags::IS_DTOR_VALID) {
```

Comment:
ah, it's pre-existing? ok, I guess we can live with it then. I thought I closed up the cells in there, but I guess not.